lib/model/CForecastModelPersist.cc

```diff
@@ -128,9 +128,9 @@ bool CForecastModelPersist::CRestore::nextModel(TMathsModelPtr& model,
                         m_ModelParams.distributionRestoreParams(dataType)},
                     m_ModelParams.distributionRestoreParams(dataType)};
 
-                if (traverser.traverseSubLevel(std::bind<bool>(
-                        maths::CModelStateSerialiser(), std::cref(params),
-                        std::ref(model_), std::placeholders::_1)) == false) {
+                auto serialiser_operator = [&params = static_cast<const maths::SModelRestoreParams&>(params), &model_](core::CStateRestoreTraverser& traverser){
```

Comment:
Please use camelCase for local variables, so `serialiserOperator`.

Also, is the cast needed?  I tried building on a Mac with clang++ and it wasn't.

I don't think the format checker will like the long line.  This worked for me with clang++ on Mac (after running through clang-format):

```
                auto serialiserOperator =
                    [&params, &model_](core::CStateRestoreTraverser& traverser_) {
                        return maths::CModelStateSerialiser()(params, model_, traverser_);
                    };
```

I also renamed the argument from `traverser` to `traverser_` to avoid a shadow warning from clang++.

If the cast is necessary with g++ then it might be better to use `const_cast` instead of `static_cast` and put it inside the lambda.  But usually a cast is not necessary to _add_ constness, only to discard constness.

Reply:
1) Interestingly, clang-format did not break the line. I did it manually now. 
2) I renamed `traverser` as you suggested. 
3) `const_cast` is used to remove `const`, not to add it, hence use of `static_cast`. I was replacing `std:cref`, hence I performed casting explicitly, but afaik it is not absolutely necessary at this point. So, I removed it.